server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -444,15 +503,15 @@ public void removeBan(TaskId parentTaskId) {
     public void applyClusterState(ClusterChangedEvent event) {
         lastDiscoveryNodes = event.state().getNodes();
         if (event.nodesRemoved()) {
-            synchronized (banedParents) {
+            synchronized (bannedParents) {
                 lastDiscoveryNodes = event.state().getNodes();
                 // Remove all bans that were registered by nodes that are no longer in the cluster state
```

Comment:
I think we can move to drop this now and rely entirely on the channel-closing trigger now. I don't think this logic ever completely worked, we could receive a task from a departing node after it had left the cluster but while it was still connected to us; moreover it might rejoin the cluster without disconnecting from us and then we'd keep on processing the task it thought it had cancelled.

(it's still needed for BWC, it's better than nothing, but should only apply to bans that aren't tracking their own channels)

Reply:
Yes, this applies to bans that are not tracked by channels (see line 512 where we check `ban.getValue().perChannel == false`).

Reply:
I will remove this on 8.0 once we backported this change to 7.x.

Reply:
Ah yes, you're right again.